db/migrate/20180107082701_authlogic_to_devise.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class AuthlogicToDevise < ActiveRecord::Migration[5.1]
+  def self.up
+    add_column :users, :unconfirmed_email, :string, limit: 254
+    add_column :users, :reset_password_token, :string
+    add_column :users, :reset_password_sent_at, :datetime
+    add_column :users, :remember_token, :string
+    add_column :users, :remember_created_at, :datetime
+    add_column :users, :authentication_token, :string
+    add_column :users, :confirmation_token, :string, limit: 255
+    add_column :users, :confirmed_at, :timestamp
+    add_column :users, :confirmation_sent_at, :timestamp
+    execute "UPDATE users SET confirmed_at = created_at, confirmation_sent_at = created_at"
+
+    rename_column :users, :password_hash, :encrypted_password
+
+    rename_column :users, :current_login_at, :current_sign_in_at
+    rename_column :users, :last_login_at, :last_sign_in_at
+    rename_column :users, :current_login_ip, :current_sign_in_ip
+    rename_column :users, :last_login_ip, :last_sign_in_ip
+    rename_column :users, :login_count, :sign_in_count
+
+    remove_column :users, :persistence_token
```

Comment:
Lets advise upgraders to take a DB dump or maybe avoid removing (just marking deprecated), as there is a chance of losing data, trying to rollback, oh, all the tokens are NULL

Reply:
The removed columns are only `:persistence_token, :single_access_token, :perishable_token` which are used for session state. Others are only renamed so rollback should be fine. Nonetheless I agree a database backup is a smart move and have made a note of it.